### PR TITLE
[flutter_plugin_android_lifecycle] Rollback the annotations version to one that is compatible

### PR DESCRIPTION
**Pull request by reidbaker**
https://github.com/flutter/flutter/issues/122039

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

### CODE REVIEW

File: packages/flutter_plugin_android_lifecycle/pubspec.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: flutter_plugin_android_lifecycle
 description: Flutter plugin for accessing an Android Lifecycle within other plugins.
 repository: https://github.com/flutter/packages/tree/main/packages/flutter_plugin_android_lifecycle
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+flutter_plugin_android_lifecycle%22
-version: 2.0.9
+version: 2.1.0
```
[the remark this answers is not in the document]
Comment (reidbaker) [reply packages/flutter_plugin_android_lifecycle/pubspec.yaml:5]: I wanted to make it the previous version but thought this was safer.

[the remark this answers is not in the document]
Comment (christopherfujino) [reply packages/flutter_plugin_android_lifecycle/pubspec.yaml:5]: it feels to me that, given this isn't really adding functionality in a backwards compatible manner, this should either be a patch bump (2.0.10) if we're saying this IS backwards compatible or a major version bump (3.0.0) if we're saying it's incompatible.

[the remark this answers is not in the document]
Comment (reidbaker) [reply packages/flutter_plugin_android_lifecycle/pubspec.yaml:5]: I agree also this comment https://github.com/flutter/packages/pull/3359#discussion_r1125512214

**Comment by stuartmorgan** [comment]
> Rollback the annotations version to one that is compatible

Compatible with what exactly?

I really dislike the pattern of rolling back published package change—affecting every client of the plugin—because flutter/flutter can't use the new version. Isn't this exactly why we have pinning in flutter/flutter in the first place? Why is the solution here not to keep the package pinned in flutter/flutter until whatever in that repo is incompatible is updated?

**Comment by reidbaker** [comment]
> > Rollback the annotations version to one that is compatible
> 
> Compatible with what exactly?
> 
> I really dislike the pattern of rolling back published package changes—affecting every client of the package—because flutter/flutter can't use the new version. Isn't this exactly why we have pinning in flutter/flutter in the first place? Why is the solution here not to keep the package pinned in flutter/flutter until whatever in that repo is incompatible is updated?

I am actually not super confident in what is happening. I believe that updating this dependency made tip of tree red. But I really dont want to roll back the update to compileSdk 33 so I am rolling back only this version update. I think this follows the spirit of not the letter of the rule rollback dont fix forward.

**Comment by stuartmorgan** [comment]
> I believe that updating this dependency made tip of tree red.

The link above is to a roller failing to roll forward (which is a pin+roller working as intended, preventing OOB breakage); @christopherfujino is anything actually red here?

> I think this follows the spirit if not the letter of the rule rollback dont fix forward.

Based on the error there it looks to me like the problem is something in flutter/flutter not being compatible with annotation 1.5 which is just blocking a roller from updating the pin of this package. Fixing that in flutter/flutter as part of unblocking the roll seems like the right solution to me, unless I am missing something about the situation.

**Comment by christopherfujino** [comment]
> > I believe that updating this dependency made tip of tree red.
> 
> The link above is to a roller failing to roll forward (which is a pin+roller working as intended, preventing OOB breakage); @christopherfujino is anything actually red here?

On the roll https://github.com/flutter/flutter/pull/121847, integration tests failed to resolve gradle dependencies. I unblocked the roller by pinning the dependency in https://github.com/flutter/flutter/pull/122043.

It sounds like fixing the framework forward and removing the pin is the best way forward, and we don't need a revert here.

**Pull request closed without merging**